shap/plots/_scatter.py

```diff
@@ -71,101 +85,107 @@ def scatter(
+    xmin, xmax, ymin, ymax : float, string, aggregated Explanation or None
+        Desired axis limits. Can be a float to specify a fixed limit.
 
-    xmax : float or string
-        Represents the upper bound of the plot's x-axis. It can be a string of the format
-        "percentile(float)" to denote that percentile of the feature's value used on the x-axis.
+        It can be a string of the format ``"percentile(float)"`` to denote that
+        percentile of the feature's value.
 
-    ax : matplotlib Axes object
-        Optionally specify an existing matplotlib ``Axes`` object, into which the plot will be placed.
-        In this case, we do not create a ``Figure``, otherwise we do.
+        It can also be an aggregated column of a single column of an :class:`.Explanation`,
+        such as ``explanation[:, "feature_name"].percentile(20)``.
+
+    overlay: dict, optional
+        Optional dictionary of curves to overlay as line plots.
+
+        The dictionary maps a curve name to a list of (xvalues, yvalues) pairs, where
+        there is one pair for each feature to be plotted.
+
+    ax : matplotlib Axes, optional
+        Optionally specify an existing matplotlib ``Axes`` object, into which
+        the plot will be placed.
+
+        Only supported when plotting a single feature.
 
     show : bool
         Whether ``matplotlib.pyplot.show()`` is called before returning.
-        Setting this to ``False`` allows the plot
-        to be customized further after it has been created.
+
+        Setting this to ``False`` allows the plot to be customized further after it
+        has been created.
+
+    Returns
+    -------
+    ax : matplotlib Axes object
+        Only returned if ``show=False``.
 
     Examples
     --------
     See `scatter plot examples <https://shap.readthedocs.io/en/latest/example_notebooks/api_examples/plots/scatter.html>`_.
 
     """
-    assert str(type(shap_values)).endswith(
-        "Explanation'>"
-    ), "The shap_values parameter must be a shap.Explanation object!"
+    if not isinstance(shap_values, Explanation):
+        raise TypeError("The shap_values parameter must be a shap.Explanation object!")
 
     # see if we are plotting multiple columns
     if not isinstance(shap_values.feature_names, str) and len(shap_values.feature_names) > 0:
+        if ax is not None:
+            raise ValueError("The ax parameter is not supported when plotting multiple features")
+        # Define order of columns (features) to plot based on average shap value
         inds = np.argsort(np.abs(shap_values.values).mean(0))
-        nan_min = np.nanmin(shap_values.values)
-        nan_max = np.nanmax(shap_values.values)
-        if ymin is None:
-            ymin = nan_min - (nan_max - nan_min) / 20
-        if ymax is None:
-            ymax = nan_max + (nan_max - nan_min) / 20
-        # FIXME: the following code ignores any passed in `ax`
```

Comment:
Yes, in a sense. When multiple columns are passed this function creates a figure with several subplots, so it's a figure-level function rather than an axes-level function and the `ax` parameter doesn't really make sense. So, now on Line ~137 we raise a ValueError if the `ax` parameter is passed when plotting multiple features.